app/controllers/homepage_controller.rb

```diff
@@ -6,5 +6,14 @@ def index
     @newbooking = Newbooking.all
     @totalbookings = @newbooking.size
     @totalmoney = @newbooking.reduce(0) { |sum, booking| sum + booking.service.price }
+    return unless params[:pet_id]
+
+    pet = Pet.find(params[:pet_id])
+    customer = pet.customer
+    make_quick_booking(pet, customer)
+  end
+
+  def make_quick_booking(pet, customer)
```

Comment:
Thinking back to when you were doing `C#`, I suspect you learned about `private` methods vs `public` methods. We have the same concept in ruby (albeit not enforced in the same way).

I *think* this is a private method as I don't believe it needs to be called from outside the controller, so it should probably be made "private".

In Ruby, we do that in one of two ways:

```ruby
class Foo
  def my_public_method
    ..
  end

  def first_private
    ...
  end

  # this changes the method above from public to private.
  # although this works.... it isn't often used because it is less obvious
  # and although worth knowing, is best avoided.
  private :first_private


  # this changes all of the methods that come after this line to be "private"
  private

  def second_private
    ...
  end
end
```